refactor(minecraft): update to 1.21.6

DataM0del opened the pull request:
~~This is going to take a *while* to even get it to compile due to `SwordItem` and `ArmorItem` being converted to data packs. I'm coming for you Mojang, get rid of data packs now!1111!111!!!!! (/j, might be useful for the sword blocking thing, but that's like the only 1 good thing about data packs).
I probably should've reopened the old PR (#5913), but yes. (best reasoning :money_mouth_face:)
Mojang will probably make everything use a data pack, so we *probably* are best off at some point trying to make everything ready for data packs, since this is annoying.~~

- [ ] It launches (before 4eeac7ad9861e30c6a59f6d956dd32b324d091dc since we're updating again)
- [ ] MCEF works (never)
- [ ] Render shortcuts / etc are fully refactored to be reimplemented
- [ ] (port specific) TODOs are gone
- [ ] Fix player slowly drifting to the left (movement input issue thing? I think I mixed it up)

DataM0del committed:
fix(liquidbounce.accesswidener): comment out removed / renamed fields

DataM0del committed:
fix(CommandItemStack): comment out creative mode check because I don't know what hasCreativeInventory is in 1.21.5

DataM0del committed:
refactor(CommandConfig): fix hover & click event errors

DataM0del committed:
fix(CommandContainers): parse -> readCompound (please work)

DataM0del committed:
fix(CommandDebug): open URL debug info error

DataM0del committed:
fix(CommandFriend): hover & click event

DataM0del committed:
fix(CommandHelp): more chat events

DataM0del committed:
fix(CommandClientIntegrationSubcommand): more chat events

DataM0del committed:
fix(CommandUsername): more chat events

DataM0del committed:
fix(CommandItemEnchant#creativeOrThrow): check if we're in creative mode as a "hack fix"

DataM0del committed:
fix(CommandItemGive, CommandItemRename, CommandItemSkull): replace `hasCreativeInventory` with more hack fixes

DataM0del committed:
fix(FakePlayer): prev{Thing} -> last{Thing}

DataM0del committed:
fix(PosPoseSnapshot): prev{Thing} -> last{Thing}

DataM0del committed:
fix(AutoClicker#isWeaponSelected): use ItemTags for checking if an item is a sword

DataM0del committed:
fix(ModuleSuperKnockback): errors

DataM0del committed:
fix(ModuleSwordBlock): use item tags for detecting sword items

screw your datapacks darn it!!!

DataM0del committed:
fix(ArmorEvaluation): slotType is a variable now

DataM0del committed:
fix(ArmorEvaluation): detekt

DataM0del committed:
fix(ArmorPiece, DisablerVanillaSpeed, SpecialItemExtension, SubmoduleIdPredict): errors

DataM0del committed:
fix: find & replace `getTickDelta` with `getTickProgress`

DataM0del committed:
fix(DisablerVulcanRiptide): `player.inventory.offHand[0]` -> `player.offHandStack`

DataM0del committed:
fix(ModuleInventoryTracker): `player.inventory.main` -> `player.inventory.mainStacks`

DataM0del committed:
fix(DisablerVerusExperimental): `player.lastBaseY` -> `player.lastY` (please work)

DataM0del committed:
fix(ModuleTeams): `getArmorStack` -> `getStack`

DataM0del committed:
fix(CustomAntiBotMode): most errors

DataM0del committed:
fix(MatrixAntiBotMode): all errors, probably broke something

DataM0del committed:
fix(ModuleDebugRecorder): click and hover event errors

DataM0del committed:
fix(ModuleNoClip#handleGameTick): `fallDistance` was changed to a double (lol)

DataM0del committed:
fix(ModuleSprint#shouldPreventSprint): use movementInput thing

DataM0del committed:
fix(FlyFireball, FlyGeneric, FlyNcpClip, LiquidWalkVulcan291, NoCheatPlusBow, SpeedGrimCollide): errors

DataM0del committed:
fix(SpeedVulcan286): goingSideways getter

DataM0del committed:
fix(SpeedVulcanGround286): isMovingSideways

DataM0del committed:
fix(AutoShopInventoryManager): errors

DataM0del committed:
todo(ItemCategorization#getItemFacets): fix these

DataM0del committed:
fix(ModuleOffhand, NoFallHypixelPacket, NoFallPacketJump, NoFallVerus, NoFallVulcan, WeaponItemFacet): errors

DataM0del committed:
fix(liquidbounce.accesswidener): uncomment & fix compute fall damage access widener

DataM0del committed:
fix(WeaponItemFacet): remove unused SwordItem import

DataM0del committed:
fix(ModuleBreadcrumbs): use VertexBuffer to draw, not sure if the GlUsage value is right

DataM0del committed:
fix(ArmorPiece): fix 1 error lol

DataM0del committed:
fix(Slots): `player.inventory.mainHandStack` -> `player.mainHandStack`

DataM0del committed:
fix(ModuleMobOwners): `entity.ownerUuid` -> `entity.owner?.uuid`

DataM0del committed:
fix(PolyglotScript, RotationExtensions): more chat events & `prev{Pitch,Yaw}` -> `last{Pitch,Yaw}`

DataM0del committed:
fix(RunnableClickEvent): add getAction (returns `RUN_COMMAND` action) function & remove superclass params

DataM0del committed:
fix(EntityExtensions): moving property & `prev{Name}` -> `last{Name}` for some things

DataM0del committed:
fix(InventoryUtils): errors

DataM0del committed:
fix(MixinClientWorld, MixinInput): got bored with fixing Kotlin errors, going on to mixins...

DataM0del committed:
fix(MixinClientPlayerEntity): 4 more errors in this file

DataM0del committed:
fix(ModuleSurround): prev{X,Y,Z} -> last{X,Y,Z}

DataM0del committed:
fix(MurderMysterySwordDetection#isSword): detect swords and pickaxes with item tags

DataM0del committed:
fix(MixinClientPlayerEntity#hookCustomMultipler): please work

DataM0del committed:
fix(ModuleHoleFiller): prev{X,Y,Z} -> last{X,Y,Z}

DataM0del committed:
fix(ModuleProjectileAimbot): added handItems

literally just taken from 1.21.4 source, it just returns a List of the main hand item and the offhand item

DataM0del committed:
fix: use player.isInCreativeMode instead of isCreative in interaction manager

Co-authored-by: ccetl <109693935+ccetl@users.noreply.github.com>

DataM0del committed:
fix(CommandConfig): use SuggestCommand instead of CopyToClipboard (oops)

DataM0del committed:
fix(MurderMysterySwordDetection): ok detekt

DataM0del committed:
fix: import handItems extension where handItems is used

DataM0del committed:
fix(ContainerFunctions): check for creative mode instead of hasCreativeInventory

DataM0del committed:
refactor(ModuleBreadcrumbs): reuse vertex buffer

DataM0del committed:
fix(RenderBufferBuilder#draw): use vertex buffer

DataM0del committed:
refactor(BlockExtensions): hasCreativeInventory() -> gameMode.isCreative()

DataM0del committed:
fix: use Gradle 8.12

DataM0del committed:
refactor: change version to 1.21.5

DataM0del committed:
fix(gradle.properties): add min_loader_version

DataM0del committed:
refactor: use Item{Stack}.is{Sword, Pickaxe, Armor}

DataM0del committed:
chore: update gradle to 8.13

DataM0del committed:
deps: update loom to 1.11.0-alpha.5

DataM0del committed:
fix: more errors :yawning_face:

DataM0del committed:
fix: more errors :yawning_face:

DataM0del committed:
fix: some find and replace

DataM0del committed:
fix: some more errors, pushing and fixing conflicts now

sqlerrorthing committed:
Merge pull request #1 from sqlerrorthing/feat/minecraft/1.21.5

chore: resolve conflicts

DataM0del committed:
fix(KillAuraRequirements): mainHandStack was moved out of inventory thing and replace `is SwordItem` with `.isSword` again

DataM0del committed:
fix: conflicts

DataM0del committed:
Merge branch 'nextgen' into feat/minecraft/1.21.5

MukjepScarlet commented on src/main/kotlin/net/ccbluex/liquidbounce/deeplearn/listener/OverlayTrainingListener.kt:
```diff
@@ -76,9 +76,8 @@ class OverlayTrainingListener(
-            mc.inGameHud.setOverlayMessage(progressBar, false)
-        }
+        // TODO: please work (removed render thread wrapping)
+        mc.inGameHud.setOverlayMessage(progressBar, false)
```
`RenderSystem.recordRenderCall` -> `mc.execute`

DataM0del replied:
:skull:
also there's more than one usage of that, but thanks

DataM0del committed:
fix: `recordRenderCall` -> `mc.execute`

DataM0del committed:
draft: tried to fix AutoArmorSaveArmor

DataM0del committed:
fix(liquidbounce.accesswidener): remove limb animator pos thing

DataM0del committed:
fix(liquidbounce.accesswidener): comment or remove broken stuff

DataM0del committed:
chore(RenderShortcuts, TargetRenderer): commented stuff out because I want to fix the mixins, not more compilation errors

DataM0del committed:
chore(TargetRenderer): commented more stuff out

DataM0del committed:
fix: more errors

DataM0del committed:
fix(ModuleClickGui): use `mc.execute`

DataM0del committed:
fix(ModuleBreadcrumbs): comment out stuff

DataM0del committed:
refactor: replace `RenderSystem.recordRenderCall {` with `mc.execute {`

DataM0del committed:
refactor: replace `RenderSystem.recordRenderCall` with `mc.execute`

DataM0del committed:
fix: more errors, this is boring, just let me fix mixins now!!!!

DataM0del committed:
fix: more errors, this is boring, just let me fix mixins now!!!!

DataM0del commented:
101 compilation errors (nice)
I can't wait for when all of these darned errors are gone and I can *finally* get to fixing the mixins...

DataM0del committed:
fix: more errors (101 errors left lol)

DataM0del commented:
probably should update the mods to the latest version too

DataM0del committed:
fix: 94 errors left & update mods

DataM0del committed:
fix: 4 errors (all in ArmorPiece)

Whoever added data packs needs to di-
I mean... happily live their life for 1 tick!
The fact that I spent 4 days just trying to fix stuff indicates that someone needs to NOT make internal changes every 5 microseconds...

DataM0del committed:
chore: merge upstream

DataM0del committed:
fix: detekt

DataM0del committed:
Merge branch 'nextgen' into feat/minecraft/1.21.5

DataM0del committed:
chore(ArmorPiece): remove some fields that caused errors because yes

DataM0del commented:
I think they got rid of a lot of blend stuff just so they can replace it with frame buffers, I have no intent of fixing the MCEF stuff...

DataM0del commented:
> I think they got rid of a lot of blend stuff just so they can replace it with frame buffers, I have no intent of fixing the MCEF stuff...

They didn't, yay!
![image](https://github.com/user-attachments/assets/7f5f23f2-ec07-490c-8866-bdd4094dd373)

DataM0del committed:
fix: it almost compiles

DataM0del commented:
also they love shuffling around `prev` to `last` in the variable names

DataM0del committed:
fix: it compiles, fails because MCEF doesn't support 1.21.5, here we go...

DataM0del committed:
fix: it launches and then crashes 2 seconds later

DataM0del committed:
fix(MixinEntity): remove fall damage fix

DataM0del committed:
fix: remove RenderSystem mixin (`clearColor` is gone)

DataM0del committed:
fix(liquidbounce.mixins.json): remove RenderSystem mixin from list

DataM0del committed:
fix(MixinCobwebBlock.java, MixinPowderSnowBlock.java): signature

DataM0del committed:
fix: 2 mixin errors

DataM0del committed:
fix: it shows the menu!!!

DataM0del committed:
fix: add useless properties to blurred UI layer so Minecraft will render it lol

DataM0del commented:
Going to work on this tomorrow.

vanvac commented:
oh my 1.21.5 changes
looks like lots of shit need to be changed

DataM0del commented:
> oh my 1.21.5 changes looks like lots of shit need to be changed

Yeah, there's like 3 new things in 1.21.5, yet all of this garbage has to be done to update LiquidBounce, so many internal changes...

DataM0del commented:
currently, as soon as the main menu (where MCEF is used) is opened, it crashes because I don't know how to make a GLTexture yet, or at least I haven't done it yet.

vanvac commented:
well idk how to do all that but take ur time cause i dont think anyone is that bodtherd about updating to 1.21.5 as it is a very small (ish) update

vanvac commented:
by not being bothered i mean like not really worried about a 1.21.5 update 1.21.4 will last for long time and many 1.9+ servers still use 1.21.1 💀

DataM0del committed:
Merge branch 'nextgen' into feat/minecraft/1.21.5

DataM0del commented:
> by not being bothered i mean like not really worried about a 1.21.5 update 1.21.4 will last for long time and many 1.9+ servers still use 1.21.1 💀

Yea, all this update really adds is the ability to customize sword blocking stuff via a data pack and misc world stuff that lots of servers LiquidBounce would be used on, *probably* won't matter for those. And ViaVersion (and consequently, VF+) supports 1.21.5 if you need to.
Also, idk how Wurst updates like instantly, we need their devs working here for porting LiquidBounce to the newest versions :money_mouth_face: (it's probably because they don't use so much rendering stuff and etc. they probably also use built-in Minecraft utils so...)

codingfire23 commented:
cool

DataM0del commented:
> cool

this PR would not be possible without:
![IMG_20250401_153158021_PORTRAIT](https://github.com/user-attachments/assets/56a078dc-2aa3-40d3-a4aa-ca407c0a48f1)

SuicidalEeper commented:
> > cool
> 
> this PR would not be possible without: ![IMG_20250401_153158021_PORTRAIT](https://private-user-images.githubusercontent.com/183248792/430204315-56a078dc-2aa3-40d3-a4aa-ca407c0a48f1.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3NTI5NTYsIm5iZiI6MTc0Mzc1MjY1NiwicGF0aCI6Ii8xODMyNDg3OTIvNDMwMjA0MzE1LTU2YTA3OGRjLTJhYTMtNDBkMy1hNGFhLWNhNDA3YzBhNDhmMS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQwNFQwNzQ0MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMTIzNzZlYzVmYjY4NGU1MzYyMzk1ODNiMjFkMzQ2Nzg4ZDVhZTk0MjNjMDE4OTA1MjhhZmE3NDk0OGIxNjEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4GdbgiWm0Yt8qOw-Yz1AseKIgb0gvYvjPPRhlNl9F80)

I LOVE

ghost commented:
What is the current state

DataM0del committed:
Merge branch 'nextgen' into feat/minecraft/1.21.5

DataM0del committed:
Merge branch 'nextgen' into feat/minecraft/1.21.5

DataM0del committed:
refactor: tried to fix JCEF tab thing

DataM0del commented:
> What is the current state

dead, hoping at some point to regain my motivation and fix MCEF lol

DataM0del commented:
for now, I'll just resolve the conflicts :money_mouth_face:

DataM0del committed:
Merge branch 'nextgen' into feat/minecraft/1.21.5

DataM0del committed:
refactor(KillAuraRequirements): fix detekt

DataM0del committed:
fix(VirtualScreenType): use Runnable & remove unused import

DataM0del committed:
fix(ModuleAutoTool): `player.inventory.mainHandStack` -> `player.mainHandStack`

DataM0del commented:
it compiles now :money_mouth_face:

DataM0del committed:
refactor(HideAppearance): hide appearance by default, so I can port stuff

DataM0del committed:
fix(MixinClientPlayerInteractionManager): hook `getSelectedSlot`

DataM0del committed:
fix: it launches (FINALLY)

DataM0del commented:
![image](https://github.com/user-attachments/assets/f7fff43a-f19a-483a-9e63-a2a965f8e05e)
YAY!!!

DataM0del committed:
refactor: don't hide appearance

vanvac commented:
Bro went from Removing lots of broken code for it to launch to it actually launch with that code fixed

DataM0del commented:
> Bro went from Removing lots of broken code for it to launch to it actually launch with that code fixed

you may notice that nothing is rendered and (you probably didn't) that I would slowly move towards a direction, even while not doing anything. those are bugs to fix
you also can't see the chat... so you're blind while trying to toggle and bind stuff if you don't have a config, console will help though

DataM0del commented:
TODO: fix mcef not working / rendering

DataM0del commented:
I'll work on that now lol, ig it's because mcef doesn't work with `GlStateManager` instead of `RenderSystem` lol

DataM0del commented:
*after fixing conflicts,* I'm going to fix MCEF now I think lol, the reason why it's not working is because MCEF generates a glId for you, and by default, the OpenGL ID for the abstract texture class is final, so I can't change it, so I'll probably use access wideners to make it mutable and update the field so the UI will actually render.

DataM0del commented:
nah, time for updating to 25w18a :money_mouth_face:

DataM0del committed:
refactor(RenderShortcuts): tried to fix stuff

DataM0del committed:
chore: merge upstream

DataM0del committed:
fix(Command, DisablerMiniblox): errors

DataM0del committed:
fix(ClientUtils, CustomAntiBotMode, ServerObserver): errors

vanvac commented:
is there a todo tracker you have setup if so approx what percent way are you through on fully updating to 1.21.5

DataM0del commented:
> is there a todo tracker you have setup if so approx what percent way are you through on fully updating to 1.21.5

no, I'll make it now

DataM0del commented:
I made it, also, I'm going to actually update it this time after I paste the lb legacy movefix into my client :money_mouth_face:

Rexjosh999 commented:
So the new update works with 1.21.5?

vanvac commented:
> So the new update works with 1.21.5?

not as of now

reading the todo on the github it only just seems to be able to launch but many stuff is still broken

DataM0del committed:
chore: merge upstream

DataM0del committed:
fix: build errors

DataM0del committed:
refactor: archivesBaseName -> archivesName

DataM0del committed:
refactor: I tried...

DataM0del commented:
1.21.6 is happening... :yawning_face:
hopefully they reverted some of the 1.21.5 changes since, well, I don't want to waste all my time trying to fix MCEF :rage:

sqlerrorthing commented:
> I don't want to waste all my time trying to fix MCEF 😡

what happends?

DataM0del commented:
> > I don't want to waste all my time trying to fix MCEF 😡
> 
> what happends?

they broke MCEF in 1.21.5 by removing some `RenderSystem` stuff and I had to replace it with `GlStateManager` stuff instead lol, also you can't tell it to register a abstract texture with a `glId` anymore, as that has been moved to `GlTexture`.
Maybe `NativeImageBackedTexture` might work?

DataM0del committed:
deps(minecraft): now update to `25w20a`...

DataM0del commented:
1.21.6 has an input packet and now uses that for sneaking... no quake 2 movement system please :pray:

DataM0del committed:
Merge branch 'CCBlueX:nextgen' into feat/minecraft/1.21.5

DataM0del committed:
fix: lots of erors

DataM0del committed:
fix: more errors

vanvac commented:
im guessing since 1.21.6 is wery close to here you are focusing your efforts on making it work with the newest version rather than keeping woriking on 1.21.5 and then updating again

DataM0del commented:
> im guessing since 1.21.6 is wery close to here you are focusing your efforts on making it work with the newest version rather than keeping woriking on 1.21.5 and then updating again

uhh... no rofl. just hoping by updating it I can fix MCEF easier on that version :skull:

DataM0del committed:
chore: tried to fix stuff (I give up with 1.21.6)

DataM0del commented:
replaced by #6303 since I'm lazy

sqlerrorthing commented:
oh no :*

vanvac commented:
replacement not complete give up

sqlerrorthing commented:
you do a lot of work man 😞

superblaubeere27 commented:
We can make use of the work anyway, so it's not all for nothing. I take a look at your implementation to get a second opinion on what the correct port for certain things should be.
